src/pinning.js

```diff
@@ -215,20 +215,26 @@ class Pinning {
     this.analytics.trackOpenDB(address, timer.timers.openDB.duration())
   }
 
-  invalidateDBCache (odbAddress, rootStoreAddress) {
+  async rewriteDBCache (odbAddress, rootStoreAddress) {
     const split = odbAddress.split('.')
     if (split[1] === 'space') {
       const spaceName = split[2]
-      this.cache.invalidate(`${rootStoreAddress}_${spaceName}`)
+      const space = await this.getSpace(rootStoreAddress, spaceName)
+      this.cache.write(`${rootStoreAddress}_${spaceName}`, space)
     } else if (split[1] === 'public') {
       // the profile is only saved under the rootStoreAddress as key
-      this.cache.invalidate(`${rootStoreAddress}`)
+      const profile = await this.getProfile(rootStoreAddress)
+      this.cache.write(rootStoreAddress, profile)
     } else if (split[1] === 'root') {
       // in this case odbAddress is the rootStoreAddress
-      this.cache.invalidate(`space-list_${odbAddress}`)
+      // const spaces = await this.listSpaces(odbAddress)
+      const spaces = []
```

Comment:
Hm, this doesn't look right?

Reply:
definitely not, its the line above, and thats why I probably should have written a new test for the one I had to comment out